tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -838,26 +829,33 @@ function Get-AzureRmProfile
     if ($ListAvailable.IsPresent)
     {
       Write-Verbose "Getting all the profiles available for install"
-      Get-ProfilesAvailable $ProfileMap
+      foreach ($profile in ($ProfileMap | get-member -MemberType NoteProperty).Name)
+      {
+        $profileObj = $ProfileMap.$profile
+        $profileObj | Add-Member -MemberType NoteProperty -Name "ProfileName" -Value $profile
+        $profileObj | Add-Member -TypeName ProfileMapData 
+        $profileObj
+      }
+      return
     }
     else
     {
       # Just display profiles installed on the machine
       Write-Verbose "Getting profiles installed on the machine and available for import"
       $IncompleteProfiles = @()
       $profilesInstalled = Get-ProfilesInstalled -ProfileMap $ProfileMap ([REF]$IncompleteProfiles)
-      $Output = @()
       foreach ($key in $profilesInstalled.Keys)
-      {
-        $Output += "Profile : $key"
-        $Output += "-----------------"
-        $Output += ($profilesInstalled.$key | Format-Table -HideTableHeaders | Out-String)
+      {      
+        $profileObj = New-Object -TypeName psobject -property $profilesinstalled.$key
+        $profileObj.PSObject.TypeNames.Insert(0,'ProfileMapData')
+        $profileObj | Add-Member -MemberType NoteProperty -Name "ProfileName" -Value $key
```

Comment:
Same question -it looks like we are pushing $profileObj to the output stream twice, is this intentional?